src/main/java/org/jabref/gui/mergeentries/newmergedialog/ThreeWayMergeView.java

```diff
@@ -74,11 +74,25 @@ private void initializeToolbar() {
         toolbar.showDiffProperty().addListener(e -> updateDiff());
         toolbar.diffViewProperty().addListener(e -> updateDiff());
         toolbar.diffHighlightingMethodProperty().addListener(e -> updateDiff());
-        this.updateDiff();
+        toolbar.hideEqualFieldsProperty().addListener(e -> showOrHideEqualFields());
+        updateDiff();
+        showOrHideEqualFields();
+    }
+
+    private void showOrHideEqualFields() {
+        for (FieldRowView row : fieldRows) {
+            if (toolbar.shouldHideEqualFields()) {
+                if (row.hasEqualLeftAndRightValues()) {
+                    row.hide();
+                }
+            } else {
+                row.show();
+            }
+        }
```

Comment:
I found it a bit hard to locate the code that shows or hides equal fields, so I moved it into its own method. It is a little slower than the old code, but since the number of fields per entry is small, it shouldn't make any difference.